shared-libs/rules-engine/test/pouchdb-provider.spec.js

```diff
@@ -249,9 +368,64 @@ describe('pouchdb provider', () => {
           chtDocs.pregnancyReport,
           reportConnectedByPlace,
         ],
-        taskDocs: [taskRequestedByChtPlace, taskRequestedByChtContact],
+        taskDocs: [
+          cancelledTaskForPlace,
+          readyTaskForPlace,
+          taskRequestedByChtPlace,
+          cancelledTask,
+          completedTask,
+          draftTask,
+          failedTask,
+          readyTask,
+          taskRequestedByChtContact
+        ],
         userSettingsId: 'org.couchdb.user:username',
       });
     });
   });
+
+  describe('allTaskRowsByOwner', () => {
+    it('should return tasks_by_contact for patient', async () => {
+      const contactIds = ['patient', 'thing'];
+
+      const rowsByOwner = await pouchdbProvider(db).allTaskRowsByOwner(contactIds);
+      chai.expect(rowsByOwner).excludingEvery('_rev').to.have.deep.members([
+        { id: completedTask._id, key: ['owner', 'all', 'patient'], value: { state: 'Completed' } },
+        { id: failedTask._id, key: ['owner', 'all', 'patient'], value: { state: 'Failed' } },
+        { id: cancelledTask._id, key: ['owner', 'all', 'patient'], value: { state: 'Cancelled' } },
+        { id: readyTask._id, key: ['owner', 'all', 'patient'], value: { state: 'Ready' } },
+        { id: draftTask._id, key: ['owner', 'all', 'patient'], value: { state: 'Draft' } },
+        { id: taskOwnedByChtContact._id, key: ['owner', 'all', 'patient'], value: { } },
+      ]);
+    });
+
+    it('should return tasks_by_contact for place', async () => {
```

Comment:
> do we want to consider db exceptions for this one and allTaskRows? thinking

Since these are integration tests (they run against an actual database), it's kind of tricky to simulate database errors, and we don't do it for the other integration tests. I'm adding a test where I pass a malformed parameter and catch the error to show that we expect errors to be thrown.